modules/augment/src/transform.cpp

```diff
@@ -0,0 +1,132 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+
+#include "precomp.hpp"
+namespace cv { namespace augment {
+
+Transform::Transform() {}
+
+Transform::~Transform() {}
+                
+void Transform::image(InputArray _src, OutputArray _dst)
+{
+    _src.copyTo(_dst);
+}
+
+Point2f Transform::point(const Point2f& src)
+{
+    return src;
+}
+
+Vec4f Transform::rectangle(const Vec4f& src)
+{
+    float x1 = src[0],
+        y1 = src[1],
+        x2 = src[2],
+        y2 = src[3];
+
+    Point2f tl(x1, y1);
+    Point2f bl(x1, y2);
+    Point2f tr(x2, y1);
+    Point2f br(x2, y2);
+
+    Point2f tl_transformed = this->point(tl);
+    Point2f bl_transformed = this->point(bl);
+    Point2f tr_transformed = this->point(tr);
+    Point2f br_transformed = this->point(br);
+
+    float x1_transformed = std::min({ tl_transformed.x, bl_transformed.x, tr_transformed.x, br_transformed.x });
+    float y1_transformed = std::min({ tl_transformed.y, bl_transformed.y, tr_transformed.y, br_transformed.y });
+    float x2_transformed = std::max({ tl_transformed.x, bl_transformed.x, tr_transformed.x, br_transformed.x });
+    float y2_transformed = std::max({ tl_transformed.y, bl_transformed.y, tr_transformed.y, br_transformed.y });
+
+    Vec4f output_box({ x1_transformed ,y1_transformed ,x2_transformed ,y2_transformed });
+    return output_box;
+}
```

Comment:
Can you briefly explain what this method does?

Reply:
We need the output bounding box to include the same object included by the original box and still represent it by 2 points, so sometimes when the box is rotated and shifted, We can get the minimum x, y and maximum x,y of the transformed points to ensure including the whole object.

This is the same approach using in data augmentation library imaug used in pytorch